docs/configuration/http_filters/cors_filter.rst

```diff
@@ -0,0 +1,61 @@
+.. _config_http_filters_cors:
+
+CORS filter
+====================
+
+This is a filter which handles Cross-Origin Resource Sharing requests based on route or virtual host settings.
+For the meaning of the headers please refer to the pages below.
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS
+https://www.w3.org/TR/cors/
+
+
+.. code-block:: json
+
+  {
+    "type": "both",
+    "name": "cors",
+    "config": {}
+  }
+
+
+Settings
+--------
+
+Settings on a route take precedence over settings on the virtual host.
+Both the route and the virtual host need to set enabled to true.
+
+.. code-block:: json
+
+  {
+    "cors": {
+        "enabled": false,
+        "allow_origin": "http://foo.example",
+        "allow_methods": "POST, GET, OPTIONS",
+        "allow_headers": "Content-Type",
+        "allow_credentials": false,
+        "expose_headers": "X-Custom-Header",
+        "max_age": "86400"
+    }
+  }
+
+enabled
+  *(optional, boolean)* Defaults to false. Setting *enabled* to false on a route disables CORS
```

Comment:
Q: Is this the most common case for this field? Or do you think we should potentially default to true? (I'm fine with whatever just want to make sure we default to the most common thing).

Reply:
Sorry! This should say defaults to true.

Reply:
Good catch!